libs/ml/include/ml/ops/top_k.hpp

```diff
@@ -0,0 +1,200 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "math/tensor.hpp"
+#include "math/top_k.hpp"
+#include "ml/exceptions/exceptions.hpp"
+#include "ml/ops/ops.hpp"
+
+#include <cassert>
+#include <utility>
+#include <vector>
+
+namespace fetch {
+namespace ml {
+namespace ops {
+
+template <class T>
+class TopK : public fetch::ml::ops::Ops<T>
+{
+public:
+  using TensorType     = T;
+  using SizeType       = typename TensorType::SizeType;
+  using TensorSizeType = fetch::math::Tensor<SizeType>;
+  using DataType       = typename TensorType::Type;
+  using ArrayPtrType   = std::shared_ptr<TensorType>;
+  using VecTensorType  = typename Ops<T>::VecTensorType;
+  using SPType         = OpTopKSaveableParams<T>;
+  using MyType         = TopK<TensorType>;
+
+  /**
+   * TopK function based on tf.top_k
+   * @param k number of k highest numbers to be outputed
+   * @param sorted TRUE=descending order, FALSE=ascending order
+   */
+  explicit TopK(SizeType k, bool sorted = true)
+    : k_(k)
+    , sorted_(sorted)
+  {}
+
+  explicit TopK(SPType const &sp)
+    : Ops<T>(sp)
+  {
+    k_      = sp.k;
+    sorted_ = sp.sorted;
+  }
+
+  ~TopK() override = default;
+
+  std::shared_ptr<OpsSaveableParams> GetOpSaveableParams() override
+  {
+    SPType sp{};
+    sp.k      = k_;
+    sp.sorted = sorted_;
+
+    return std::make_shared<SPType>(sp);
+  }
+
+  std::shared_ptr<fetch::ml::ops::Ops<TensorType>> MakeSharedCopy(
```

Comment:
this is inside fetch::ml::ops namespace already right? no need for explicit namespacing then?